### PR TITLE
upgrade to polkadot-v0.9.40

### PR DESCRIPTION
**Pull request by cuteolaf**
- [x] Upgrade dependencies

### CODE REVIEW

**Comment by cuteolaf** [comment]
@S0AndS0 @Rubberbandits @shibshib @eduardogr 
A quick question: How long does it usually take to generate weights for `subtensor` pallet?
Based on my understanding of the benchmarking scripts, I assume quite a long time for benchmarking `set_weight`

**Comment by Rubberbandits** [comment]
> @S0AndS0 @Rubberbandits @shibshib @eduardogr A quick question: How long does it usually take to generate weights for `subtensor` pallet? Based on my understanding of the benchmarking scripts, I assume quite a long time for benchmarking `set_weight`

I switched set_weight benchmarking to burned registration from PoW, which significantly speeds up benchmarking. Only takes about a minute now.

File: runtime/src/lib.rs
Comment (Rubberbandits) [inline runtime/src/lib.rs:1]: This config for GRANDPA finality gadget seems wrong

Comment (cuteolaf) [reply runtime/src/lib.rs:1]: Please let me know what the issue is and feel free to correct it.

**Comment by cuteolaf** [comment]
> > @S0AndS0 @Rubberbandits @shibshib @eduardogr A quick question: How long does it usually take to generate weights for `subtensor` pallet? Based on my understanding of the benchmarking scripts, I assume quite a long time for benchmarking `set_weight`
> 
> I switched set_weight benchmarking to burned registration from PoW, which significantly speeds up benchmarking. Only takes about a minute now.

one minute for generating all the weights?

**Comment by Rubberbandits** [comment]
> > > @S0AndS0 @Rubberbandits @shibshib @eduardogr A quick question: How long does it usually take to generate weights for `subtensor` pallet? Based on my understanding of the benchmarking scripts, I assume quite a long time for benchmarking `set_weight`
> > 
> > 
> > I switched set_weight benchmarking to burned registration from PoW, which significantly speeds up benchmarking. Only takes about a minute now.
> 
> one minute for generating all the weights?

That's correct, though I cant remember if its merged into main yet.

**Comment by cuteolaf** [comment]
seems like it's not merged into main yet.

**Comment by cuteolaf** [comment]
which branch is it? I'd like to cherry-pick the necessary commits.

**Comment by Rubberbandits** [comment]
> which branch is it? I'd like to cherry-pick the necessary commits.

https://github.com/opentensor/subtensor/blob/eaa2c82e39f3dd5084143599788d6bf4d3cccbb7/pallets/subtensor/src/benchmarks.rs

It's apart of the sudo decentralization PR, which is waiting for python CLI to be complete as the chain side is finished (barring some APIs i have to expose for proposal data)

**Pull request closed without merging**